src/main/java/hlf/java/rest/client/service/impl/NetworkStatusImpl.java

```diff
@@ -288,25 +316,63 @@ public ResponseEntity<ClientResponseModel> commitChannelConfigTransaction(
-                    addOrgToChannelWriteSetBuilder.buildWriteset(readset, organizationDetails))
-                .build();
-        MessageOrBuilder message = configUpdate;
-        String channelConfigString = JsonFormat.printer().print(message);
+        ConfigUpdate configUpdate = createConfigUpdate(channelName, organizationDetails);
+        String channelConfigString = JsonFormat.printer().print(configUpdate);
+        log.info(channelConfigDeserialization.deserializeValueFields(channelConfigString));
+        UpdateChannelConfiguration updateChannelConfiguration = new UpdateChannelConfiguration();
+        updateChannelConfiguration.setUpdateChannelConfiguration(
+            configUpdate.toByteString().toByteArray());
+        selectedChannel.updateChannelConfiguration(
+            updateChannelConfiguration,
+            selectedChannel.getUpdateChannelConfigurationSignature(
+                updateChannelConfiguration, user));
+      } catch (InvalidArgumentException e) {
+        log.warn(
+            "Error while committing channel config: One or more arguments included in the config update are invalid");
+        throw new ServiceException(
+            ErrorCode.HYPERLEDGER_FABRIC_TRANSACTION_ERROR,
+            "One or more arguments included in the config update are invalid",
+            e);
+      } catch (TransactionException e) {
+        log.warn("Error while committing channel config: " + e.getMessage());
+        throw new FabricTransactionException(
+            ErrorCode.HYPERLEDGER_FABRIC_TRANSACTION_ERROR,
+            ErrorCode.HYPERLEDGER_FABRIC_TRANSACTION_ERROR.name(),
+            e);
+      } catch (IOException e) {
+        log.warn("Error while establishing connection to the gateway");
+        throw new ServiceException(
+            ErrorCode.HYPERLEDGER_FABRIC_CONNECTION_ERROR,
+            "Error while establishing connection to the gateway",
+            e);
+      }
+      return new ResponseEntity<>(
+          new ClientResponseModel(ErrorConstants.NO_ERROR, ErrorCode.SUCCESS.getReason()),
+          HttpStatus.OK);
+    } else {
+      log.warn("Network and User cannot be NULL: " + "Network = " + network + "and User = " + user);
+      return new ResponseEntity<>(
+          new ClientResponseModel(
+              ErrorCode.NOT_FOUND.getValue(),
+              "Network and User cannot be NULL: " + "Network = " + network + "and User = " + user),
+          HttpStatus.OK);
+    }
+  }
+
+  @Override
+  public ResponseEntity<ClientResponseModel> addAnchorPeersToChannel(
+      String channelName, ChannelUpdateParamsDTO channelUpdateParamsDTO) {
+    Network network = gateway.getNetwork(channelName);
+    if (network != null && user != null) {
+      try {
+        Channel selectedChannel = network.getChannel();
+        ConfigUpdate configUpdate = createConfigUpdate(channelName, channelUpdateParamsDTO);
+        String channelConfigString = JsonFormat.printer().print(configUpdate);
```

Comment:
[question] The method `getDeserializedConfig` appears to be doing this task with the exception handling.

Reply:
The `getDeserializedConfig` method returns the Base64 encoded characters. Here we are interested in converting a protobuf message to JSON format, that's why didn't reuse the method.